compiler/qsc_eval/src/output.rs

```diff
@@ -4,14 +4,40 @@
 use std::io::{Cursor, Write};
 
 use num_bigint::BigUint;
-use num_complex::Complex64;
+use num_complex::{Complex, Complex64};
 
 #[derive(Copy, Clone, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct Error;
 
 #[must_use]
 pub fn format_state_id(id: &BigUint, qubit_count: usize) -> String {
-    format!("|{:0>qubit_count$}⟩", id.to_str_radix(2))
+    format!("|{}⟩", fmt_basis_state_label(id, qubit_count))
+}
+
+#[must_use]
+pub fn get_phase(c: &Complex<f64>) -> f64 {
+    f64::atan2(c.im, c.re)
+}
+
+#[must_use]
+pub fn fmt_complex(c: &Complex<f64>) -> String {
+    // Format -0 as 0
+    // Also using Unicode Minus Sign instead of ASCII Hyphen-Minus
+    // and Unicode Mathematical Italic Small I instead of ASCII i.
+    format!(
+        "{}{:.4}{}{:.4}𝑖",
+        if c.re <= -0.00005 { "−" } else { "" },
+        c.re.abs(),
+        if c.im <= -0.00005 { "−" } else { "+" },
+        c.im.abs()
+    )
+}
+
+#[must_use]
+pub fn fmt_basis_state_label(id: &BigUint, qubit_count: usize) -> String {
+    // This will generate a bit string that shows the qubits in the order
+    // of allocation, left to right.
```

Comment:
Wait, isn't it right to left? (I know I wrote the comment but I was probably confused). At least the playground does right to left, I don't know if that one uses a different formatting ([example](https://microsoft.github.io/qsharp/?code=H4sIAAAAAAAACn1Ty44aMRC8z1e0xGWQEGg5BkVKNqxEDiTL4wNoTA9jxS9sj1gU7b%2Bn7YGdgSiZC7hdXV1VtieTCQxgg9opKia8eCalYBMx5mUuDWBOQXjporSmA4UECmA9vLyuwaH0AdATBEdCVlLAqUETG%2F0BrCCeLVf3MoZME2uM4Ml5CmQiL7lZJiUhQonmABrfpEY1BHrLCjPJjZZb0BwVaf4z%2FhC7rWWA1QCct0ePGjJfgrC4nu5xYVBTcCjo6h5%2BF8CfdWRgKYW3wVZxvGqHjb%2Bb6KUJUsz%2Bh5pLPBobohRhVmTglxduvLxaaWI5vLV6TElmNTnpcgifoFxTaFQcQfs7vOpJH9v6qpQVDM0hdTG2EZ4l29oTNIEOEC0ITwl67%2FdGxiA4PcFnWCWGcji735n2d%2FoKNtSeUMXnHNvxIA2Ehv04G2S2tL%2BAQKWkOWbsbrHr%2FI6gT3eupagBnVOS0tEs8IAaPev3aEJlvW5DYjuJKc8b9wm2NTHl7RI85tKEDw3ffvzc9mR0JIvy9NSzn3BcGXEGj9Zre25vZ4oyX%2BR%2FTps32i1R1NLQDqrGiPuZve3HgJeEofF%2FOUnvID2QnL7mY%2BZsMuiSX5snxX187pgBfULizqvWPd%2BeX50KxWw63YHlfQS5Pm3r0159nebfQ6%2Blx6jWFBtv8kzd%2BklvLxngOx06Bb7FlZrz1rdR78V78QfJ0BmQkQQAAA%3D%3D)) .

Reply:
@minestarks The comment is C&P as is. This matches the playground.